src/umpire/ResourceManager.cpp

```diff
@@ -118,52 +145,105 @@ ResourceManager::initialize()
 #endif
 
 #if defined(UMPIRE_ENABLE_DEVICE)
-  m_memory_resources[resource::Device] = registry.makeMemoryResource("DEVICE", getNextId());
+  {
+    const std::string name = resource::type_to_string(resource::Device);
+    m_memory_resources[name] = registry.makeMemoryResource(name, getNextId());
+  }
 #endif
 
 #if defined(UMPIRE_ENABLE_PINNED)
-  m_memory_resources[resource::Pinned] = registry.makeMemoryResource("PINNED", getNextId());
+  {
+    const std::string name = resource::type_to_string(resource::Pinned);
+    m_memory_resources[name] = registry.makeMemoryResource(name, getNextId());
+  }
 #endif
 
 #if defined(UMPIRE_ENABLE_UM)
-  m_memory_resources[resource::Unified] = registry.makeMemoryResource("UM", getNextId());
+  {
+    const std::string name = resource::type_to_string(resource::Unified);
+    m_memory_resources[name] = registry.makeMemoryResource(name, getNextId());
+  }
 #endif
 
 #if defined(UMPIRE_ENABLE_CUDA)
-  m_memory_resources[resource::Constant] = registry.makeMemoryResource("DEVICE_CONST", getNextId());
+  {
+    const std::string name = resource::type_to_string(resource::Constant);
+    m_memory_resources[name] = registry.makeMemoryResource(name, getNextId());
+  }
 #endif
 
   /*
    * Construct default allocators for each resource
    */
-  auto host_allocator = m_memory_resources[resource::Host];
-  m_allocators_by_name["HOST"] = host_allocator;
-  m_allocators_by_id[host_allocator->getId()] = host_allocator;
+  {
+    const std::string name = resource::type_to_string(resource::Host);
+    auto host_allocator = m_memory_resources[name];
+    m_allocators_by_name[name] = host_allocator;
+    m_allocators_by_id[host_allocator->getId()] = host_allocator;
 
-  m_default_allocator = host_allocator;
+    m_default_allocator = host_allocator;
+  }
+
+#if defined(UMPIRE_ENABLE_NUMA_HOST)
+  {
+    const std::string base_name = "NUMA_NODE_";
+    auto host_nodes = numa::get_host_nodes();
+    for (std::size_t n : host_nodes) {
+      resource::MemoryResourceTraits traits;
+      traits.numa_node = n;
+      const std::string name = base_name + std::to_string(n);
+      m_memory_resources[name] = registry.makeMemoryResource(name, getNextId(), traits);
+    }
+  }
+#endif
 
 #if defined(UMPIRE_ENABLE_DEVICE)
```

Comment:
I don't think we want to create these if ENABLE_NUMA is on. For each physical resource, we should only support one way of accessing it.

Reply:
If we remove this and keep everything else untouched, then allocators on NUMA nodes, other than the preferred node one replacing the malloc allocator for resource::Host, will not be found. `getAllocatorFor()` loops through `m_memory_resources` here:
```c++
Allocator
ResourceManager::getAllocatorFor(const resource::MemoryResourceTraits traits)
{
  UMPIRE_LOG(Debug, "(Looking up allocator by traits)");

  const resource::MemoryResourceTraits d_traits;
  for (auto r : m_memory_resources) {
```

Reply:
With the new option naming, my question is: what happens if ENABLE_NUMA_DEVICES is on? We don't want to also make the cudaMalloc device?